pkg/operator/ceph/object/spec.go

```diff
@@ -60,10 +77,26 @@ func (c *config) startDaemonset() error {
 			Template: c.makeRGWPodSpec(),
 		},
 	}
-	k8sutil.SetOwnerRefs(c.context.Clientset, c.store.Namespace, &daemonset.ObjectMeta, c.ownerRefs)
+	k8sutil.SetOwnerRefs(c.context.Clientset, c.store.Namespace, &d.ObjectMeta, c.ownerRefs)
+
+	logger.Debugf("starting rgw daemonset: %+v", d)
+	_, err := c.context.Clientset.ExtensionsV1beta1().DaemonSets(c.store.Namespace).Create(d)
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create rgw daemonset %s: %+v", c.instanceName(), err)
+		}
+		logger.Infof("daemonset for rgw %s already exists. updating if needed", c.instanceName())
+		// There may be a *lot* of rgws, and they are stateless, so don't bother waiting until the
```

Comment:
DaemonSets also have rolling updates similar to deployments

Reply:
@travisn If we don't use them yet we should enable them so the update happens automatically after a Kubernetes defined policy and not with the user needing to manually delete Pods. :slightly_smiling_face: 
Though enabling this update policy for the DaemonSet may be seen out of scope I would like to see it in the PR as it should hopefully "just" be a few lines like this (but in Go code):
```
apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
   name: zk
spec:
   serviceName: zk-hs
   replicas: 3
   updateStrategy:
       type: RollingUpdate
```
(see https://kubernetes.io/blog/2017/09/kubernetes-statefulsets-daemonsets/)

Reply:
Yes, we could certainly set the update strategy if the default is not sufficient.

Reply:
@travisn Mhh I could have sworn the default is is `OnDelete`. Maybe in the "old" Extensions APIs and it was changed in the new/latest apps/v1 Deployments.
See https://kubernetes.io/docs/tasks/manage-daemon/update-daemon-set/#daemonset-update-strategy.

Reply:
I believe what you are talking about is already set above. Or am I missing something?
```go
Spec: extensions.DaemonSetSpec{
 			UpdateStrategy: extensions.DaemonSetUpdateStrategy{
 				Type: extensions.RollingUpdateDaemonSetStrategyType,
 			},
			Template: c.makeRGWPodSpec(),
		},
```